secure_uninstall/module.py

```diff
@@ -0,0 +1,81 @@
+# coding: utf-8
+##############################################################################
+#
+#    Copyright (C) All Rights Reserved 2014 Akretion
+#    @author David BEAL <david.beal@akretion.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from openerp.osv import orm, fields
+from openerp.tools.config import config
+
+
+class Module(orm.Model):
+    _inherit = 'ir.module.module'
+
+    def button_uninstall(self, cr, uid, ids, context=None):
+        if 'uninstall_authorized' in context:
+            del context['uninstall_authorized']
```

Comment:
I'm not sure this will work if the new API comes into play, with context being a frozen dict. You should make a copy of context before modifying it. 

Also, check for context being None, and set it to `{}`  before your test for the key